### DISCUSSION TITLE
Seccure connections

### DISCUSSION THREAD

**Discussion by Vagg-eng** [Ericsson/ecchronos#785] [General]
Hello,
We tried to integrate ecChronos in VMs.
Does ecChronos supports secure connections and how certificates  management is perform?
How ecChronos is aware what certificates to use?
How it is informed in case of certificate expiration?

Thanks a lot for helping
///Vaggelis

**Comment by tommystendahl** [comment]
Hi,
ecchronos supports certificates, in security.yml we have the option to pass a path for the certificates in jmx and cql, so ecchronos mounts the cql session and JmxConnector with theses credentials. For the web server these options can be found in the application.yml.
Certificates are validated at connection. If the certificate expires, the session keeps alive until its restarted.

**Comment by 0rlych1kk4** [comment]
Hi, Thanks for the clarification.
Just to add an operational perspective for deployments running ecChronos in long-lived VM environments.
Since certificate validation happens only at connection time and existing sessions remain active after expiration, it may be worth planning certificate lifecycle handling outside of ecChronos. In practice, teams typically:
- Monitor certificate expiry proactively (external checks or PKI monitoring)
- Align certificate renewal with a controlled ecChronos restart
- Use automation (e.g., rolling restarts) after certificate rotation
For environments with strict security policies or short-lived certificates, this helps avoid situations where expired credentials remain active longer than intended.
As a potential enhancement, it could be useful to optionally expose certificate validity status via logs or metrics, or provide a warning when certificates are close to expiration.

Thanks for the work on ecChronos.
